### PR TITLE
Check VID/PID when doing device attestation

### PR DESCRIPTION
**Pull request by cecille**
#### Problem
Device attestation currently uses a hardcoded vid/pid, not the one read from the basic cluster

#### Change overview
- Adds VID/PID check to device attestation.

#### Testing
- VID/PID are now as expected, commissioning passes on linux

### CODE REVIEW

**Review by msandstedt — APPROVED** [review]

File: src/controller/CHIPDeviceController.cpp
Comment (bzbarsky-apple) [inline src/controller/CHIPDeviceController.cpp:1730]: It might be worth reading all of the Basic bits we care about plus the network commissioning feature maps in s single read, using the same setup as #14410 is doing.  Just instead of a single wildcard path, pass in whatever list paths we care about, some of which might be wildcards...  Then wait for the read cache to get back to us, and get the values from that.  It'd probably cut down significantly on the round-trips.

Comment (cecille) [reply src/controller/CHIPDeviceController.cpp:1730]: I agree, but I got that network cluster stuff working at 11:00 friday night...sliding in under the deadline like Indiana jones (if we stretch a bit what it means to hit the deadlines). I agree that would be preferable, but at this point I think I would need approval to put it in.

Comment (bzbarsky-apple) [reply src/controller/CHIPDeviceController.cpp:1730]: I think my point is that once #14410 merges, we should consider using its machinery here to just get all that data in a single read....  

That does not need to block this PR, but can happen as a folowup.  And we can figure out the process as needed, I expect.

Comment (cecille) [reply src/controller/CHIPDeviceController.cpp:1730]: sounds good. https://github.com/project-chip/connectedhomeip/issues/14669 to track

File: src/credentials/DeviceAttestationVerifier.h
Comment (tcarmelveilleux) [inline src/credentials/DeviceAttestationVerifier.h:218]: Let's create an issue to to refactor this argument list into a single POD structure as a follow-up since this is becoming much too large of an argument list

**Review by tcarmelveilleux — APPROVED** [review]

File: src/controller/AutoCommissioner.cpp
Comment (chrisdecenzo) [inline src/controller/AutoCommissioner.cpp:101]: Will this be a problem for a device on ethernet (or that otherwise gets it creds using means outside Matter)? For example, I'm not sure it makes sense for a tv casting phone app to try to fake out these network commissioning cluster features

**Review by chrisdecenzo — APPROVED** [review]

File: src/controller/AutoCommissioner.cpp
Comment (bzbarsky-apple) [reply src/controller/AutoCommissioner.cpp:101]: This part is not reached for on-network commissioning.  As in, if we discovered the device over dns-sd already, this part will not matter.

**Review by msandstedt — APPROVED** [review]

File: src/credentials/DeviceAttestationVerifier.h
Comment (cecille) [reply src/credentials/DeviceAttestationVerifier.h:218]: https://github.com/project-chip/connectedhomeip/issues/14758

**Comment by cecille** [comment]
Please do not submit until I PSA the platform devs about this change.

**Pull request merged**
